src/RequestParser.cpp

```diff
@@ -0,0 +1,537 @@
+#include "RequestParser.hpp"
+#include "error.hpp"
+
+/* ====== HELPER FUNCTIONS ====== */
+
+/**
+ * @brief Checks if a given string starts with a single space followed by a non-space character.
+ *
+ * This function checks if the provided string `str` starts with exactly one space character followed
+ * by a non-space character. If this condition is met, it returns a substring of `str` starting from 
+ * the second character. If the condition is not met, it sets the error code to 400 and throws a 
+ * `std::runtime_error` with an appropriate error message.
+ *
+ * @param str The input string to be checked.
+ * @return A substring of `str` starting from the second character if the input string starts with
+ *         a single space followed by a non-space character.
+ * @throws std::runtime_error If the input string does not start with a single space followed by a 
+ *         non-space character, an error is thrown and the error code is set to 400.
+ */
+std::string	RequestParser::checkForSpace(const std::string& str)
```

Comment:
I'm not saying that we should eliminate the check. Rather combine them into one.
If the string is not "GET ", i.e. GET and then a singular space it would be wrong.
Advantage would be less context switch since we already compare the first characters for the method name, so we could just look one character further?
But we can also leave it separate, your call.

Reply:
I think it is cleaner this way. Every function should only perform one task.

Reply:
Ok